test/instructions/opt_not_test.rb

```diff
@@ -4,8 +4,80 @@
 
 class TenderJIT
   class OptNotTest < JITTest
-    def test_opt_not
-      skip "Please implement opt_not!"
+    def not_false
+      !false
+    end
+
+    def test_opt_not_false
+      jit.compile method(:not_false)
+      assert_equal 1, jit.compiled_methods
+      assert_equal 0, jit.executed_methods
+
+      jit.enable!
+      v = not_false
+      jit.disable!
+      assert_equal true, v
```

Comment:
I was going to suggest using `assert v` but I think it is reasonable to have it explicitly expecting `true` to avoid asserting some wrong but truthy value